.github/workflows/python-ci.yml

```diff
@@ -0,0 +1,46 @@
+#
+# This workflow will build, run tests, and pack for the Microsoft.TeamsAI .NET SDK.
+#
+
+name: python-ci
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - 'python/**'
+
+jobs:
+  python-build-test-lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Dependencies
+        working-directory: ./python
+        run: |
+          python -m pip install --upgrade pip
+          pip install poetry
+          poetry install
+      - name: Check Dependencies
+        working-directory: ./python
+        run: |
+          poetry check
+      - name: Lint
```

Comment:
Not sure whether we should split some checks into separate job, like Lint. So CI can list all the errors instead of stop at a certain step. You can make the decision.

Reply:
I usually like to keep the commands separate to make the CI easier to read, ie a different command for each item, but still within the same job. I am open to creating different jobs for each tho so long as we have the .Net and JS projects do the same. Right now this matches what the C# team is doing.